### PR TITLE
[ECMS-7472]Document Content: exo fields: exo:title crashed by exo: name

### PR DESCRIPTION
**Pull request by AtiaZ**
Description : 
- keep the existing behavior only for the exo:name property.

### CODE REVIEW

File: core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by exo on 9/22/16.
+ */
+public class TestDialogFormUtil extends TestCase {
```
Comment (thomasdelhomenie) [inline core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java]: You don't need to extend TestCase when you use annotations (@Test).

File: core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  private static final String name = "name";
+  private static final String title = "title";
+
+  public void setUp() throws Exception {
```
Comment (thomasdelhomenie) [inline core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java]: This is useless.

File: core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    properties.put(name, jcrExoName);
+    properties.put(title, jcrExoTitle);
+    Map<String, JcrInputProperty> map = DialogFormUtil.prepareMap(inputs, properties, null);
+    assertNotNull(map);
+    assertEquals(jcrExoName.getValue(), "name");
+    assertEquals(jcrExoTitle.getValue(), "title");
+  }
+
+
+  public void tearDown() throws Exception {
```
Comment (thomasdelhomenie) [inline core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java]: Useless

File: core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  private static final String name = "name";
+  private static final String title = "title";
+
+  public void setUp() throws Exception {
+    super.setUp();
+  }
+
+  @Test
+  public void testPrepareMap() throws Exception {
+    List inputs = new ArrayList<UIFormStringInput>();
```
Comment (thomasdelhomenie) [inline core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java]: Try to structure your tests with the Given/When/Then pattern, it is more readable IMO. For example : https://github.com/exodev/commons/blob/develop/commons-file-storage/src/test/java/org/exoplatform/commons/file/services/FileSystemBinaryProviderTest.java#L53-L69

File: core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    jcrExoName.setJcrPath("/node/exo:name");
+    JcrInputProperty jcrExoTitle = new JcrInputProperty();
+    jcrExoTitle.setJcrPath("/node/exo:title");
+    properties.put(name, jcrExoName);
+    properties.put(title, jcrExoTitle);
+    Map<String, JcrInputProperty> map = DialogFormUtil.prepareMap(inputs, properties, null);
+    assertNotNull(map);
+    assertEquals(jcrExoName.getValue(), "name");
+    assertEquals(jcrExoTitle.getValue(), "title");
+  }
```
Comment (thomasdelhomenie) [inline core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java]: We should add another test to test the case where title is null IMO.

File: core/webui/pom.xml [hunk 6]
```diff
@@ -138,5 +138,24 @@
...
+      <artifactId>junit</artifactId>
+      <scope>test</scope>
+    </dependency>
   </dependencies>
+  <build>
+  <pluginManagement>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
```
Comment (thomasdelhomenie) [inline core/webui/pom.xml]: Is it plugin management block really useful ?

Comment (AtiaZ) [inline core/webui/pom.xml]: plugin removed.

File: core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java (hunk 3 above)
Comment (AtiaZ) [inline core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java]: removed

File: core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java (hunk 5 above)
Comment (AtiaZ) [inline core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java]: done

File: core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java (hunk 2 above)
Comment (AtiaZ) [inline core/webui/src/test/java/org/exoplatform/ecm/webui/TestDialogFormUtil.java]: removed

File: core/webui/pom.xml [hunk 7]
```diff
@@ -138,5 +138,14 @@
       <artifactId>javax.servlet-api</artifactId>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>org.exoplatform.ecms</groupId>
+      <artifactId>ecms-core-webui-explorer</artifactId>
```
Comment (thomasdelhomenie) [inline core/webui/pom.xml]: ecms-core-webui-explorer depends on ecms-core-webui, so by adding a dependency from ecms-core-webui to ecms-core-webui-explorer you introduce a cyclic dependency. It should be removed.

Comment (AtiaZ) [inline core/webui/pom.xml]: removed and use UIDialogForm instead of UIDocumentForm in the Unit Test

**Comment by thomasdelhomenie** [comment]
LGTM
